harness/determined/common/storage/base.py

```diff
@@ -159,3 +162,26 @@ def _list_directory(root: Union[str, os.PathLike]) -> Dict[str, int]:
                 result[rel_path] = os.path.getsize(abs_path)
 
         return result
+
+
+def from_string(shortcut: str) -> StorageManager:
+    p: urllib.parse.ParseResult = urllib.parse.urlparse(shortcut)
+    if any((p.params, p.query, p.fragment)):
+        raise ValueError(f'Malformed checkpoint_storage string "{shortcut}"')
+    scheme = p.scheme.lower()
+    if scheme in ["", "file"]:
+        base_path = p.path
+        return storage.SharedFSStorageManager(base_path=base_path)
+    elif scheme == "gs":
+        bucket = p.netloc
+        prefix = p.path.lstrip("/")
+        return storage.GCSStorageManager(bucket=bucket, prefix=prefix)
+    elif scheme == "s3":
+        bucket = p.netloc
+        prefix = p.path.lstrip("/")
+        return storage.S3StorageManager(bucket=bucket, prefix=prefix)
+    else:
+        raise ValueError(
+            f'Could not understand storage manager scheme "{p.scheme}". Use "gs", "s3", "file", '
+            f"or omit it. "
```

Comment:
Avoid f-strings without any `{}` inside of them.  Actually I thought one of our linters cared about this...

Reply:
side note, let's rephrase "or omit it", because while omitting the scheme is allowed, a confused user reading this might not realize that "omitting it" means "the shortcut string must then be interpreted as a plain path".

Maybe: `Supported schemes are: "gs", "s3", or "file", or you may also provide a path to a local directory"`